File: src/main/java/com/squarepolka/digiscope/plot/plotpoint/PlotPointBinary.java
package com.squarepolka.digiscope.plot.plotpoint;


import java.math.BigDecimal;

public class PlotPointBinary extends PlotPointPulse {

    private boolean digitalValue;
    private BigDecimal digitLength;

    public PlotPointBinary(PlotPointPulse plotPointPulse, BigDecimal digitLength, boolean digitalValue) {
        super(plotPointPulse.getStartPulse(), plotPointPulse.getEndPulse());
        this.digitLength = digitLength;
        this.digitalValue = digitalValue;
    }

    public String toString() {
        if (digitalValue) {
            return super.toString() + "\tbinary 1 (length " + digitLength + ")";
        } else {
            return super.toString() + "\tbinary 0 (length " + digitLength + ")";
        }
    }

    public boolean isDigitalValue() {
        return digitalValue;
    }

}

File: src/main/kotlin/plotpoint/PlotPointBinary.kt
package plotpoint

import java.math.BigDecimal

class PlotPointBinary (val plotPointPulse: PlotPointPulse,
                       val digitLength: BigDecimal,
                       val digitalValue: Boolean): PlotPointPulse(plotPointPulse.getStartPulse(), plotPointPulse.getEndPulse()) {

    override fun toString(): String {
        return if (digitalValue) {
            super.toString() + "\tbinary 1 (length " + digitLength + ")"
        } else {
            super.toString() + "\tbinary 0 (length " + digitLength + ")"
        }
    }

}